scripts/miniapps.py

```diff
@@ -0,0 +1,128 @@
+from os import system, makedirs
+from os.path import expanduser
+from re import sub
+
+# Finds two factors of `n` that are as close to each other as possible.
+#
+# Note: the second factor is larger or equal to the first factor
+def _sq_factor(n):
+    for i in range(1, int(n ** 0.5) + 1):
+        if n % i == 0:
+            f = (i, n // i)
+    return f
+
+
+def _check_ranks_per_node(rpn):
+    if not (rpn == 1 or rpn == 2):
```

Comment:
I don't understand why the choice of ranks per node is constrained.
Depending on the architecture (e.g. multiple numa nodes or multiple gpu per node might require more).